src/test/kotlin/uk/gov/justice/digital/hmpps/prisonregister/service/PrisonServiceTest.kt

```diff
@@ -342,4 +342,19 @@ class PrisonServiceTest {
         .isInstanceOf(ContactDetailsNotFoundException::class.java)
     }
   }
+
+  @Nested
+  inner class SetEmailAddress {
+
+    private val prisonRepository2: PrisonRepository = mock()
```

Comment:
why not use the one in https://github.com/ministryofjustice/prison-register/blob/73bf03288eaa67c63b15796cd0c141bf2a1073d0/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonregister/service/PrisonServiceTest.kt#L47 ?

Reply:
I have got different behaviour when I checked KOVER report, inner class had 2 tests 
I am not sure if in the context or anything the mock repo got alterated when I was using same mock , that is why I decided to use a second mock which worked well.

Reply:
all the other tests use _prisonRepository_, it if doesn't work it would be better to understand why

Reply:
I have added a commit to reset repository mocks https://github.com/beforeeach this fix the issue of reusing them. this apply to all 3 PR opened with similar tests.